### PR TITLE
ci: reorg docker images

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

we mount home directory and cargo directory when we run docker. actually we can just install those dependencies into our docker image.

#### Summary of Changes

build our Docker images on Ubuntu.

(will upload these images when I get approve. I have already test them and all green 🤞 but I will still run them with our ci agents again to ensure everything is fine)

### CODE REVIEW

**Comment by yihau** [comment]
btw, this one is a breaking change. will ping everyone when this one got merge.

File: .buildkite/hooks/post-checkout [hunk 1]
```diff
@@ -38,10 +38,3 @@ source ci/env.sh
     kill -9 "$victim" || true
   done
 )
-
-# HACK: These are in our docker images, need to be removed from CARGO_HOME
-#  because we try to cache downloads across builds with CARGO_HOME
-# cargo lacks a facility for "system" tooling, always tries CARGO_HOME first
-cargo uninstall cargo-audit &>/dev/null || true
-cargo uninstall svgbob_cli &>/dev/null || true
-cargo uninstall mdbook &>/dev/null || true
```
Comment (t-nelson) [inline .buildkite/hooks/post-checkout:47]: :raised_hands:

**Review by t-nelson — APPROVED** [review]
lgtm, thanks!

i wonder if it'd make sense to eventually break out some of the `RUN` stuff to separate scripts that we can share with bare metal ci machine setup, etc

**Comment by yihau** [comment]
> i wonder if it'd make sense to eventually break out some of the RUN stuff to separate scripts that we can share with bare metal ci machine setup, etc

oh yeah! I think it's a good idea. If we have those scripts, we can simply copy and run them when building the Docker image. btw, I'm considering moving all tests to Docker. That way, maybe no one will need to set up their CI machines; they'll just need to install Docker 😈

**Review by t-nelson — APPROVED** [review]

**Comment by yihau** [comment]
@t-nelson could you have a look at these 3 following PRs (https://github.com/solana-labs/solana/pull/33942, https://github.com/solana-labs/solana/pull/33943, https://github.com/solana-labs/solana/pull/33944) by any chance 🙏

**Pull request merged**